keela-exe/cameracontrolwindow.cpp

```diff
@@ -305,41 +330,102 @@ void Keela::CameraControlWindow::remove_split_frame_ui() {
 	}
 }
 
+int Keela::CameraControlWindow::init_aravis_controller() {
+	return camera_manager->init_aravis_controller();
+}
+
 void Keela::CameraControlWindow::update_gain_range() {
 	// get the range supported by the camera hardware
 	auto gain_range = camera_manager->get_gain_range();
 	double min_gain = gain_range.first;
 	double max_gain = gain_range.second;
+	// and the current gain to set the initial spin value
+	auto current_gain = camera_manager->get_gain();
 
-	if(min_gain == 0.0 && max_gain == 0.0) {
+	if(std::isnan(min_gain)) {
 		gain_spin.m_spin.set_sensitive(false);
 		spdlog::warn("Gain control not supported by camera - disabling gain control UI");
 		return;
 	}
-	// update the gain spin with the new range
+	// update the gain spin with the new range and current setting
+	gain_signal.block();
 	gain_spin.m_spin.set_adjustment(Gtk::Adjustment::create(min_gain, min_gain, max_gain, 1.0));
+	gain_spin.m_spin.set_value(current_gain);
 	gain_spin.m_spin.set_sensitive(true);
+	gain_signal.unblock();
 
 	spdlog::info("Updated gain control range to {:.1f} - {:.1f} dB", min_gain, max_gain);
 }
 
 void Keela::CameraControlWindow::update_exposure_time_range() {
 	// get the range supported by the camera hardware
-	auto exposure_time_range = camera_manager->get_exposure_time_range();
-	double min_exposure_time = exposure_time_range.first;
-	double max_exposure_time = exposure_time_range.second;
+	auto [min_exposure_time, max_exposure_time] = camera_manager->get_exposure_time_range();
+	// and the current exposure time to set the initial spin value
+	auto current_exposure_time = camera_manager->get_exposure_time();
 
-	if(min_exposure_time == 0.0 && max_exposure_time == 0.0) {
+	if(std::isnan(min_exposure_time)) {
 		exposure_time_spin.m_spin.set_sensitive(false);
-		spdlog::warn(
-		    "Exposure time control not supported by camera - disabling "
-		    "exposure time control UI");
+		spdlog::warn("Exposure time control not supported by camera - disabling exposure time control UI");
 		return;
 	}
-	// update the exposure time spin with the new range
+	// update ui based on hardware capabilities and current setting
+	exposure_time_signal.block();
 	exposure_time_spin.m_spin.set_adjustment(
 	    Gtk::Adjustment::create(min_exposure_time, min_exposure_time, max_exposure_time, 1000.0));
+	exposure_time_spin.m_spin.set_value(current_exposure_time);
 	exposure_time_spin.m_spin.set_sensitive(true);
+	exposure_time_signal.unblock();
 
 	spdlog::info("Updated exposure time control range to {:.1f} - {:.1f} μs", min_exposure_time, max_exposure_time);
 }
+
+void Keela::CameraControlWindow::update_binning_modes() {
```

Comment:
perhaps the binning spin could be a standalone Widget? This should clean up the `cameracontrolwindow` a bit